klayout/drc/testing/run_regression.py

```diff
@@ -0,0 +1,924 @@
+
+    if len(klayout_v_list) < 1 or len(klayout_v_list) > 3:
+        logging.error("Was not able to get klayout version properly.")
+        exit(1)
+    elif len(klayout_v_list) == 2:
+        if klayout_v_list[1] < 28:
+            logging.warning("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
+            )
+            exit(1)
+    elif len(klayout_v_list) == 3:
+        if klayout_v_list[1] < 28 :
+            logging.warning("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
+            )
+            exit(1)
+
+
+def get_switches(yaml_file, rule_name):
+    """Parse yaml file and extract switches data
+    Parameters
+    ----------
+    yaml_file : str
+            yaml config file path given py the user.
+    Returns
+    -------
+    yaml_dic : dictionary
+            dictionary containing switches data.
+    """
+
+    # load yaml config data
+    with open(yaml_file, 'r') as stream:
+        try:
+            yaml_dic = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    switches = list()
+    for param, value in yaml_dic[rule_name].items():
+        switch = f"{param}={value}"
+        switches.append(switch)
+
+    return switches
+
+
+def parse_results_db(results_database):
+    """
+    This function will parse Klayout database for analysis.
+
+    Parameters
+    ----------
+    results_database : string or Path object
+        Path string to the results file
+
+    Returns
+    -------
+    set
+        A set that contains all rules in the database with violations
+    """
+
+    mytree = ET.parse(results_database)
+    myroot = mytree.getroot()
+
+    # Initial values for counter
+    rule_counts = defaultdict(int)
+
+    for z in myroot[7]:
+        rule_name = f"{z[1].text}".replace("'", "")
+        rule_counts[rule_name] += 1
+
+    return rule_counts
+
+
+def run_test_case(
+    drc_dir,
+    layout_path,
+    run_dir,
+    table_name,
+):
+    """
+    This function run a single test case using the correct DRC file.
+
+    Parameters
+    ----------
+    drc_dir : string or Path
+        Path to the location where all runsets exist.
+    layout_path : stirng or Path object
+        Path string to the layout of the test pattern we want to test.
+    run_dir : stirng or Path object
+        Path to the location where is the regression run is done.
+    table_name : string
+        Table name that we are running on.
+
+    Returns
+    -------
+    dict
+        A dict with all rule counts
+    """
+
+    # Initial value for counters
+    rule_counts = defaultdict(int)
+
+    # Get switches used for each run
+    sw_file = os.path.join(Path(layout_path.parent).absolute(), f"{table_name}.{SUPPORTED_SW_EXT}")
```

Comment:
would also advise using https://docs.python.org/3/library/pathlib.html here.

Reply:
Is that a must?

Reply:
Created a separate issue for that https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/issues/5